fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -268,38 +340,12 @@ private TwoLevelContentCache() {
                     .recordStats()
                     .removalListener(((key, value, cause) -> {
                         LOG.debug(key + " to be eliminated to disk, reason: " + cause);
-                        if (isExistOnDisk(key)) {
+                        if (isExistOnDiskCache(key)) {
                             LOG.debug(key + " has cached on disk");
                             return;
                         }
-                        HadoopOutputFile tmpOutputFile = (HadoopOutputFile) IOUtil.getTmpOutputFile(
-                                METADATA_CACHE_DISK_PATH, key);
-                        PositionOutputStream outputStream = tmpOutputFile.createOrOverwrite();
-                        Path localFilePath = new Path(IOUtil.remoteToLocalFilePath(METADATA_CACHE_DISK_PATH, key));
-                        try {
-                            for (ByteBuffer buffer : value.buffers) {
-                                outputStream.write(buffer.array());
-                            }
-                            outputStream.close();
-                            if (!tmpOutputFile.getFileSystem().rename(tmpOutputFile.getPath(), localFilePath)) {
-                                LOG.warn("failed on rename: {} to {}", tmpOutputFile.getPath().toString(),
-                                        localFilePath.toString());
-                                tmpOutputFile.getFileSystem().delete(tmpOutputFile.getPath(), false);
-                            } else {
-                                diskCache.put(key, new DiskCacheEntry(value.length,
-                                        IOUtil.getOutputFile(localFilePath).toInputFile()));
-                            }
-                        } catch (IOException e) {
-                            LOG.warn("failed on writing file to disk" + e.getMessage());
-                            try {
-                                if (outputStream != null) {
-                                    outputStream.close();
-                                }
-                                tmpOutputFile.getFileSystem().delete(tmpOutputFile.getPath(), false);
-                            } catch (IOException ioException) {
-                                LOG.warn("failed on deleting file :" + tmpOutputFile.getPath());
-                            }
-                        }
+                        this.getDiskCacheEntry(key, k -> DiskCacheEntry.newDiskCacheEntry(
+                                ByteBufferInputStream.wrap(value.buffers), value.length, key));
```

Comment:
Could value be null？

Reply:
this is evicted from memCache, it shouldn't be null.